src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -749,10 +752,95 @@ namespace OpenLoco::World::MapGenerator
     {
-        registers regs;
-        regs.ebp = X86Pointer(buildingObj);
-        regs.ebx = id;
-        call(0x0042E893, regs);
+        for (auto& industry : IndustryManager::industries())
+        {
+            auto* industryObj = ObjectManager::get<IndustryObject>(industry.objectId);
+            if (!industryObj->hasFlags(IndustryObjectFlags::requiresElectricityPylons))
+                continue;
+
+            auto numTownsLeft = 3;
+            for (auto& town : TownManager::towns())
+            {
+                // Figure out distance and rotation (angle) between town and industry
+                auto rotationBL = 2;
+                auto xDist = town.x - industry.x; // ax
+                if (xDist < 0)
+                {
+                    xDist = -xDist;
+                    rotationBL = 0;
+                }
+
+                auto rotationBH = 1;
+                auto yDist = town.y - industry.y; // dx
+                if (yDist < 0)
+                {
+                    yDist = -yDist;
+                    rotationBH = 3;
+                }
+
+                auto totalDist = xDist + yDist; // bp
+                if (totalDist < 800 || totalDist > 2240)
+                    continue;
+
+                // Figure out the preferred rotation and distance to work with
+                if (yDist > xDist)
+                {
+                    std::swap(xDist, yDist);
+                    std::swap(rotationBL, rotationBH);
+                }
+
+                // 0x0042E91B
+                for (auto attemptsLeft = 2; attemptsLeft > 0; attemptsLeft--)
+                {
+                    // 0x0042E932
+                    auto remainingDistance = xDist;
+                    auto preferredRotation = rotationBL;
+
+                    // 0x0042E945
+                    auto targetNumber = remainingDistance - buildingObj->averageNumberOnMap;
+                    if (targetNumber < 0)
+                    {
+                        rotationBL = rotationBH; // bl = bh
+                        yDist = xDist;           // dh = dl
+                        continue;
+                    }
+
+                    // 0x0042E956
+                    auto index = (rotationBH << 8) | rotationBL;
+                    auto offset = World::Pos2(_503C6C[index].x * targetNumber, _503C6C[index].y * targetNumber);
+                    auto targetPos = World::Pos2(industry.x, industry.y) + offset;
+                    if (!drawableCoords(targetPos))
+                    {
+                        continue;
+                    }
+
+                    // 0x0042E98A
+                    auto tile = TileManager::get(targetPos);
+                    auto* surface = tile.surface();
+                    if (surface == nullptr)
+                    {
+                        continue;
+                    }
+
+                    auto baseZ = TileManager::getSurfaceCornerHeight(*surface);
```

Comment:
Try stick to the word `height` when referring to things in bigZ. `baseZ` is generally used for smallZ.